tv_apps/smart-system/js/app_authentication_dialog.js

```diff
@@ -202,6 +209,42 @@
       this._event = null;
     };
 
+  /**
+   * Determine whether the dialog is visible.
+   * @memberof AppAuthenticationDialog.prototype
+   */
+  AppAuthenticationDialog.prototype.isVisible =
+    function aad_isVisible() {
+      return this.element && this.element.classList.contains('visible');
+    };
+
+  /**
+   * Get z-index of the dialog.
+   * @memberof AppAuthenticationDialog.prototype
+   * @return {Number} z-index of the dialog
+   */
+  AppAuthenticationDialog.prototype.getOrder =
+    function aad_getOrder() {
+      if (this.element) {
+        var zIndex = window.getComputedStyle(this.element).zIndex;
+        return zIndex === 'auto' ? 0 : zIndex;
+      }
+    };
+
+  /**
+   * Focus the cancel button of authentication dialog
+   * @memberof AppAuthenticationDialog.prototype
+   */
+  AppAuthenticationDialog.prototype.focus =
+    function aad_focus() {
+      if (this.elements && this.elements.httpAuthenticationCancel) {
+        setTimeout(function() {
+          document.activeElement.blur();
+          this.elements.httpAuthenticationCancel.focus();
```

Comment:
Is it possible to focus the first input box? We should focus it because this is an input form. I feel that the default focus should be the first input box.